src/coreclr/jit/assertionprop.cpp

```diff
@@ -2663,6 +2663,59 @@ AssertionIndex Compiler::optAssertionIsSubtype(GenTree* tree, GenTree* methodTab
     return NO_ASSERTION_INDEX;
 }
 
+//------------------------------------------------------------------------------
+// optVNStrengthReductionOnTree: Substitutes tree with a less expensive tree using VN
+//
+// Arguments:
+//    block  -  The block containing the tree.
+//    parent -  The parent node of the tree.
+//    tree   -  The tree node
+//
+// Return Value:
+//    Returns a new tree or nullptr if nothing is changed.
+//
+GenTree* Compiler::optVNStrengthReductionOnTree(BasicBlock* block, GenTree* parent, GenTree* tree)
```

Comment:
Name seems a bit odd -- doesn't immediately look related to strength reduction. Maybe pick something like `optVNRemoveRedundantCast`?

Reply:
I am ok with either `optVNBasedFold` or `optVNBasedMorph`. cc @dotnet/jit-contrib @AndyAyersMS @SingleAccretion for thoughts 🙂